packages/components/NativeRadioButton/macos/RadioButton.swift

```diff
@@ -0,0 +1,21 @@
+class RadioButton: NSButton {
+
+	@objc public var onPress:RCTBubblingEventBlock?
+	
+	@objc public func callOnPress() {
+		self.window?.makeFirstResponder(self)
+		if (onPress != nil) {
+			onPress!(nil)
+		}
+	}
+	
+	/// Set button state to 'on' when it becomes focused
+	override func becomeFirstResponder() -> Bool {
+		/// fire onChange event
+		if (state != NSButton.StateValue.on) {
+			callOnPress()
+		}
```

Comment:
should we override the state property and call `callOnPress()` inside didSet ?

Reply:
Probably not because the callback function shouldn't get triggered when the default state is On.